src/API/TaskService.h

```diff
@@ -16,55 +16,170 @@
 #include "boost/date_time/special_defs.hpp" //include all types plus i/o
 #include <optional>
 
+/*
+ * public API for adding tasks to the Model,adding subtasks to tasks, getting tasks, getting subtasks.
+ * API can postpone tasks, delete and complete them.
+ * Also it provides the capability of representing information in the sorted way. Current sorting options:
+ * sort tasks by label, priority, date
+ * @see GeneralView
+ * @see Task
+ *
+ * @author Ilya Ilyin
+ */
 class TaskService {
 
-public:
-    bool
-    addTask(std::string taskName, boost::gregorian::date date = boost::gregorian::date(boost::gregorian::min_date_time),
-            Task::Priority priority = Task::Priority::NONE, std::string label = "");
-
-    bool addSubTaskToParent(TaskDTO &parent, std::string taskName,
-                            boost::gregorian::date = boost::gregorian::date(boost::gregorian::min_date_time),
-                            Task::Priority priority = Task::Priority::NONE, std::string label = "");
-
-    bool addTask(Task &task);
-
-    bool addSubTaskToParent(TaskDTO &parent, Task &task);
-
-    std::optional<TaskDTO> getTask(TaskID id);
-
-    std::optional<std::vector<TaskDTO>> getSubtasks(TaskID id);
-
-public:
-    bool postponeTask(TaskDTO &task, boost::gregorian::date = boost::gregorian::date(boost::gregorian::min_date_time));
-
-    bool deleteTask(TaskDTO &task);
-
-    bool completeTask(TaskDTO &task);
-
-public:
-    std::vector<TaskDTO> showAllByPriority();
-
-    std::vector<TaskDTO> showAllByLabel();
-
-    std::vector<TaskDTO> showAllByDate();
-
-    std::vector<TaskDTO> showTodayByPriority();
-
-    std::vector<TaskDTO> showTodayByLabel();
-
-    std::vector<TaskDTO>
-    showDueDateByPriority(boost::gregorian::date = boost::gregorian::date(boost::gregorian::min_date_time));
-
-    std::vector<TaskDTO>
-    showDueDateByLabel(boost::gregorian::date = boost::gregorian::date(boost::gregorian::min_date_time));
-
-    std::vector<TaskDTO>
-    showDueDateByDate(boost::gregorian::date = boost::gregorian::date(boost::gregorian::min_date_time));
-
-private:
-    AllDataStorage storage_;
-    dtoConvertor dto_convertor_;
+ public:
+  /*
+   * adds task to the model. All params are params to construct task
+   * @see Task
+   *
+   * @param taskName representing task name
+   * @param date representing date to complete task. Set to boost::gregorian::min_date_time by default
+   * @param priority Task Priority. Set to NONE by default
+   * @param label Task label. Set empty by default
```

Comment:
You won't add links on Task & Priority classes? :)

Reply:
priority class is already in Task